src/backend/cdb/cdbmutate.c

```diff
@@ -366,6 +376,46 @@ get_partitioned_policy_from_flow(Plan *plan)
 									   GP_POLICY_DEFAULT_NUMSEGMENTS());
 }
 
+static bool
```

Comment:
Before implementing this walkers (or before changing mutators to walkers if it makes sense) we had a talk about nested motions, about separate plan branches, plan back tracking. I want to see a brief comment about why this walkers are enough for our case.

Reply:
@dnskvlnk, provide answers for questions in this PR

Reply:
> We talked about mutators, not walkers. The difference is that mutators create a copy of a plan, so if you try to change an original node, this won't affect the plan, as the copy of this node will be created. Unlike mutators, walkers keep the original tree, so any changes made with walkers will be saved.

But how this explain my question?
What I'm asking:
>  I want to see a brief comment about why this walkers are enough for our case.

I want to see an explanation of why we don't have nested motions, separate plan branches, so thus, why we don't need a complex solution based on plan back tracking, and why can use simple walkers.

Reply:
We considered broadcast motions only, but later we figured out that it's unlikely to have more than one broadcast motion in a plan, so we don't need any nesting. Therefore, we don't need any nesting, backtracking or anything else. It's enough to have a simple walker which will make a single traversal

Reply:
Can you add a brief to code?